### ISSUE TITLE
Pistol crosshairs not displaying correctly

### ISSUE THREAD

**Issue by ddulude95** [ValveSoftware/halflife#3463]
It seems to be that the 9mmhandgun and 357 crosshair sprites are not displaying correctly in any resolution I try. At first I thought it was a 1440p widescreen scaling issue, but even switching to 800x600 or 1024x768 the crosshairs still are not displaying correctly. See my example screenshot. And even at one point last night, I was able to fix it by switching away from 1440p, going to any other resolution, then going back to 1440p and it would display correctly. That wasn't fixing it today.

This is on a clean "half-life/valve" folder (renamed existing folder and then verified game files)

![crosshairs_bug](https://github.com/ValveSoftware/halflife/assets/20364616/c71e893f-917c-465f-a32a-0d4e17dad807)

**Comment by darkreader2636** [comment]
By looking at sprite files this is sure a rendering issue 
![image](https://github.com/ValveSoftware/halflife/assets/119101878/62fd476f-c763-48c8-b564-e372720bb85f)

**Comment by ddulude95** [comment]
Yeah it definitely is, I checked the sprites as well and the files are fine. Not sure what's happening for them to be rendered like this. Only seems to be the pistol and 357 as well. Everything else including hud and other weapon crosshairs seems fine

**Comment by ph1sh55** [comment]
Perhaps it may have to do with this old bug valve introduced long ago? - https://tfcresource.teamfortressclassic.org/steam_era_blurry_crosshair_fix.html
You may want to check if gl_texturemode "GL_NEAREST_MIPMAP_LINEAR" changes anything?

**Comment by SirYodaJedi** [comment]
`gl_texturemode` doesn't affect sprites that aren't part of the 3D scene; use `gl_spriteblend` instead.

**Comment by ph1sh55** [comment]
> `gl_texturemode` doesn't affect sprites that aren't part of the 3D scene; use `gl_spriteblend` instead.

perhaps it should not- but the reality is it absolutely does, at least in Team Fortress Classic.  You can clearly see the crosshair clarity difference between those modes when loading TFC for the last ~20 years. (examples shown in the link I included).

**Comment by ddulude95** [comment]
> > `gl_texturemode` doesn't affect sprites that aren't part of the 3D scene; use `gl_spriteblend` instead.
> 
> perhaps it should not- but the reality is it absolutely does, at least in Team Fortress Classic. You can clearly see the crosshair clarity difference between those modes when loading TFC for the last ~20 years. (examples shown in the link I included).

I tried the fix listed on that TFC website, but it changed nothing regarding the crosshairs unfortunately.

**Comment by vektor451** [comment]
this might be 8 months old, but looking at the sprite files for the crosshair personally i've found out that the crosshairs are actually misaligned in the file and aren't centered. if you center the crosshairs yourself it seems to work perfectly. might not work for your case, but it worked for me.

**Comment by Gema2k** [comment]
I don't know if I'm breaking a rule about github posts, but changing the value in the last line of "crosshair" from 24 to 23 seems to have corrected it slightly. (I assume it also applies to the scope's default value of 2560 by reducing one number less)

**Comment by ddulude95** [comment]
> this might be 8 months old, but looking at the sprite files for the crosshair personally i've found out that the crosshairs are actually misaligned in the file and aren't centered. if you center the crosshairs yourself it seems to work perfectly. might not work for your case, but it worked for me.

How did you end up centering it? I opened it up in photoshop myself and it seems there's no way to center it since the dimensions dont allow for it to be perfectly centered

**Comment by vektor451** [comment]
> How did you end up centering it? I opened it up in photoshop myself and it seems there's no way to center it since the dimensions dont allow for it to be perfectly centered

I'm pretty sure i just moved it up and to the left by 1 pixel.
